src/pages/tezos/customizebot.js

```diff
@@ -0,0 +1,332 @@
+import React, { Suspense, useRef, useState, useEffect } from 'react';
+import NavBar from '../../components/NavBar';
+import Button from '../../components/Buttons';
+
+import { Canvas, useFrame } from 'react-three-fiber';
+import {
+  ContactShadows,
+  Environment,
+  useGLTF,
+  OrbitControls,
+  Html,
+} from '@react-three/drei';
+
+import { proxy, useProxy } from 'valtio';
+import { HexColorPicker } from 'react-colorful';
+import 'react-colorful/dist/index.css';
+import GLTFExporter from 'three-gltf-exporter';
+
+// TODO: Make every mesh part colorable -- dependent on how the material is named inside blender
+const state = proxy({
+  current: null,
+  items: {
+    head: '#ffffff',
+    body: '#ffffff',
+    hands: '#ffffff',
+    legs: '#ffffff',
+  },
+});
+
+function useGroup(scene, type) {
+  const result = [];
+
+  const filterType = [type];
+  const regexType = new RegExp(filterType.join('|'), 'i');
+
+  scene.children.forEach(group => {
+    if (regexType.test(group.name)) {
+      result.push(group);
+    }
+  });
+
+  // console.log('result', result);
+  return result;
+}
+
+const renderGroup = (groupObject, id = 0, color, color_name) => {
+  return (
+    <>
+      <group
+        name="bot_head"
+        position={groupObject.length > 0 && groupObject[id].position}
+        rotation={groupObject.length > 0 && groupObject[id].rotation}
+        scale={groupObject.length > 0 && groupObject[id].scale}
+      >
+        {groupObject.length > 0 &&
+          groupObject[id].children.map(child => {
+            child.material.name = color_name;
+            return (
+              <mesh
+                name={child.name}
+                material={child.material}
+                geometry={child.geometry}
+                position={child.position}
+                scale={child.scale}
+                material-color={color}
+              />
+            );
+          })}
+      </group>
+    </>
+  );
+};
+
+const Bot = ({ headCount, legCount, bodyCount }) => {
+  const group = useRef();
+  const { scene } = useGLTF('cryptobot.glb');
```

Comment:
@harsh242  Where is the cryptobot.glb file located?

Reply:
Locally I have it inside the public folder.